packages/core/test/Form_test.js

```diff
@@ -1524,6 +1524,12 @@ describeRepeated("Form common", createFormComponent => {
         });
         Simulate.submit(node);
         sinon.assert.notCalled(onError);
+        sinon.assert.calledWithMatch(onSubmit.lastCall, {
+          errors: [],
+          errorSchema: {},
+          schemaValidationErrors: [],
+          schemaValidationErrorSchema: {},
+        });
       });
```

Comment:
Let's add a UI test like https://github.com/rjsf-team/react-jsonschema-form/blob/739a303d5ac59c7ef50b149c56ad777a6044116a/packages/core/test/Form_test.js#L630-L634 that tests the exact scenario for the user that this PR fixes.

Reply:
Added new UI tests as requested checking for generated HTML